Add note on the danger of closed indices (#61332)
elastic/elasticsearch: docs/reference/indices/open-close.asciidoc

A closed index can be opened which will then go through the normal
recovery process.

> IMPORTANT: {es} will not rebuild any copies of shards of closed indices that

Reviewer: Because there's a risk of data loss, I'd use the `WARNING` admon here.
```suggestion
WARNING: {es} will not rebuild any copies of shards of closed indices that
```

> are lost when a node leaves your cluster. This means you will should expect to

Reviewer: Extra word.
```suggestion
are lost when a node leaves your cluster. This means you should expect to
```

> IMPORTANT: {es} will not rebuild any copies of shards of closed indices that
> are lost when a node leaves your cluster. This means you will should expect to
> lose the data held in closed indices if you replace the nodes in your cluster
> over time. Because of the risk of data loss it is best to avoid keeping closed
> indices in your cluster for a long period of time. You should instead prefer to
> <<frozen-indices,freeze>> indices that you want to keep available for
> occasional searches with low overhead, and store any other indices outside the
> cluster using a <<modules-snapshots,snapshot>>.

Reviewer: This warning is fine as-is, but I'd reorder it a bit to lead with the recommendation
and follow up with the _why_.

If you accept this suggestion, you can ignore the other ones.
```suggestion
[WARNING]
====
To reduce the risk of data loss, avoid keeping closed indices for long periods.
When a node leaves your cluster, {es} does not rebuild shard replicas for closed
indices. If you regularly replace nodes in your cluster, this will eventually
result in data loss for the cluster's closed indices.

If you want to reduce the overhead of an index while keeping it available for
occasional searches, <<frozen-indices,freeze>> the index instead. If you want
to store an index outside of the cluster, use a <<modules-snapshots,snapshot>>.
====
```

`action.destructive_requires_name` flag in the config file to `true`.
This setting can also be changed via the cluster update settings api.

Closed indices consume a significant amount of disk-space which can cause

> problems in managed environments. Closing indices can be disabled via the
> cluster settings API by setting `cluster.indices.close.enable` to `false`. The
> default is `true`.

Reviewer: Reorders this para a little.
```suggestion
problems in managed environments. Closed indices are enabled by default. You can
disable closed indices by setting `cluster.indices.close.enable` to `false`
using the <<cluster-update-settings,cluster settings API>>.
```
